pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -595,13 +613,24 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 
 		if getPodRevision(replicas[target]) != updateRevision.Name || !isHealthy(replicas[target]) {
 			unavailablePods = append(unavailablePods, replicas[target].Name)
-		} else if completedErr := inplaceupdate.CheckInPlaceUpdateCompleted(replicas[target]); completedErr != nil {
+		} else if completedErr := inplaceupdate.CheckInPlaceUpdateCompleted(
+			replicas[target]); completedErr != nil {
 			klog.V(4).Infof("StatefulSet %s/%s check Pod %s in-place update not-ready: %v",
 				set.Namespace,
 				set.Name,
 				replicas[target].Name,
 				completedErr)
 			unavailablePods = append(unavailablePods, replicas[target].Name)
+		} else {
+			// check if the updated pod is running and available given minReadySeconds
+			if isAvailable, waitTime := isRunningAndAvailable(replicas[target], minReadySeconds); !isAvailable {
+				// the pod is not available yet given the minReadySeconds requirement
+				unavailablePods = append(unavailablePods, replicas[target].Name)
+				// make sure that
+				if waitTime != 0 {
+					durationStore.Push(getStatefulSetKey(set), waitTime)
```

Comment:
Note that this is within a loop so the waitTime should the mininal waitTime across all unavailable Pods.  The durationStore.Push should happen outside of the loop.

Reply:
The loop is short-circuited by the return so we can't put the push outside of the loop but I will use the min time.